### PR TITLE
Revive fixes - part 1

### PR DESCRIPTION
**Pull request by zak-pawel**
**[Revive](https://github.com/mgechev/revive)** is fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint.

This is part1 of fixes for problems found for following set of rules:
- [rule.blank-imports]
  - Disallows blank imports -> Blank import should be only in a main or test package, or have a comment justifying it.
- [rule.context-as-argument]
  - `context.Context` should be the first argument of a function. -> By [convention](https://github.com/golang/go/wiki/CodeReviewComments#contexts), `context.Context` should be the first parameter of a function. This rule spots function declarations that do not follow the convention.
- [rule.dot-imports]
  - Forbids `.` imports. -> Importing with `.` makes the programs much harder to understand because it is unclear whether names belong to the current package or to an imported package. More information [here](https://github.com/golang/go/wiki/CodeReviewComments#import-dot)
- [rule.error-strings]
  - Conventions around error strings. -> Description: By convention, for better readability, error messages should not be capitalized or end with punctuation or a newline. More information [here](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)
- [rule.error-naming]
  - Naming of error variables. -> By convention, for the sake of readability, variables of type `error` must be named with the prefix `err`.
- [rule.indent-error-flow]
  - Prevents redundant else statements. -> To improve the readability of code, it is recommended to reduce the indentation as much as possible. This rule highlights redundant else-blocks that can be eliminated from the code. More information [here](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow)
- [rule.errorf]
  - Should replace `errors.New(fmt.Sprintf())` with `fmt.Errorf()` -> It is possible to get a simpler program by replacing `errors.New(fmt.Sprintf())` with `fmt.Errorf()`. This rule spots that kind of simplification opportunities.

### CODE REVIEW

File: plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go [hunk 1]
```diff
@@ -298,16 +298,15 @@ func (m *OpenConfigTelemetry) collectData(ctx context.Context,
...
-						// Retry with delay. If delay is not provided, use default
-						if m.RetryDelay.Duration > 0 {
-							m.Log.Debugf("Retrying %s with timeout %v", grpcServer,
-								m.RetryDelay.Duration)
-							time.Sleep(m.RetryDelay.Duration)
-							continue
-						} else {
-							return
-						}
+						return
```
Comment (srebhan) [inline plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go:309]: I think you can also strip the `else` here.

File: plugins/outputs/amqp/amqp.go [hunk 2]
```diff
@@ -296,22 +296,22 @@ func (q *AMQP) publish(key string, body []byte) error {
...
-			if err != nil {
-				return nil, err
-			}
+	}
+
+	var buf bytes.Buffer
+	for _, metric := range metrics {
+		octets, err := q.serializer.Serialize(metric)
+		if err != nil {
+			log.Printf("D! [outputs.amqp] Could not serialize metric: %v", err)
```
Comment (srebhan) [inline plugins/outputs/amqp/amqp.go:305]: That output should really use the telegraf logger, but that's probably another issue. Should we open one?

Comment (zak-pawel) [reply plugins/outputs/amqp/amqp.go:305]: No, I will change it here :)

**Review by srebhan — COMMENTED** [review]
Wow that was a ride. :-) 
The code looks good to me. I found one more superfluous `else` in `plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go` line 308. Please fix this one also.

I also found `plugins/outputs/amqp/amqp.go` using the `log` package instead of `telegraf.Logger` but that might be a separate bunch of changes...

File: plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go]: Done

File: plugins/outputs/amqp/amqp.go (hunk 2 above)
Comment (zak-pawel) [reply plugins/outputs/amqp/amqp.go]: "log" packages removed from files which were changed in previous commit. I replaced it by `telegraf.Logger` when it was easy, hard cases need to wait...

**Review by srebhan — APPROVED** [review]
Perfect! Thank you very much!

File: plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go [hunk 3]
```diff
@@ -261,7 +261,7 @@ func (c *CiscoTelemetryMDT) handleTelemetry(data []byte) {
 	msg := &telemetry.Telemetry{}
 	err := proto.Unmarshal(data, msg)
 	if err != nil {
-		c.acc.AddError(fmt.Errorf("Cisco MDT failed to decode: %v", err))
+		c.acc.AddError(fmt.Errorf("cisco MDT failed to decode: %v", err))
```
Comment (ssoroka) [inline plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go:264]: This is an interesting case. Cisco is a proper name, and should be capitalized. We could reorder the error message?

File: plugins/inputs/postgresql_extensible/postgresql_extensible.go [hunk 4]
```diff
@@ -8,7 +8,7 @@ import (
 	"strings"
 	"time"
 
-	_ "github.com/jackc/pgx/stdlib"
+	_ "github.com/jackc/pgx/stdlib" //to register stdlib from PostgreSQL Driver and Toolkit
```
Comment (ssoroka) [inline plugins/inputs/postgresql_extensible/postgresql_extensible.go:11]: I think we'll find this linter rather annoying.

File: plugins/outputs/amqp/amqp.go [hunk 5]
```diff
@@ -3,7 +3,6 @@ package amqp
 import (
 	"bytes"
 	"fmt"
-	"log"
```
Comment (ssoroka) [inline plugins/outputs/amqp/amqp.go:6]: we might want our own linter that tells users not to import log into plugins.

File: plugins/outputs/librato/librato.go [hunk 6]
```diff
@@ -17,12 +17,13 @@ import (
...
 type Librato struct {
-	APIUser   string `toml:"api_user"`
-	APIToken  string `toml:"api_token"`
-	Debug     bool
-	SourceTag string // Deprecated, keeping for backward-compatibility
-	Timeout   internal.Duration
-	Template  string
+	APIUser   string            `toml:"api_user"`
+	APIToken  string            `toml:"api_token"`
+	Debug     bool              `toml:"debug"`
```
Comment (ssoroka) [inline plugins/outputs/librato/librato.go:22]: not part of your change, but I'm suspicious of this variable now

File: plugins/outputs/wavefront/wavefront.go [hunk 7]
```diff
@@ -142,7 +142,7 @@ func (w *Wavefront) Connect() error {
 			FlushIntervalSeconds: flushSeconds,
 		})
 		if err != nil {
-			return fmt.Errorf("Wavefront: Could not create Wavefront Sender for Url: %s", w.Url)
+			return fmt.Errorf("wavefront: Could not create Wavefront Sender for Url: %s", w.Url)
```
Comment (ssoroka) [inline plugins/outputs/wavefront/wavefront.go:145]: Wavefront is a proper name and shouldn't be lowercased, but it's also not necessary as the logger will tag the plugin name elsewhere

**Review by ssoroka — APPROVED** [review]
Amazing effort. minor feedback.

File: plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go (hunk 3 above)
Comment (zak-pawel) [reply plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go]: Message modified.

File: plugins/inputs/postgresql_extensible/postgresql_extensible.go (hunk 4 above)
Comment (zak-pawel) [reply plugins/inputs/postgresql_extensible/postgresql_extensible.go:11]: It is pretty valid rule, should be used for special cases only - I needed to add a comment only in 7 places.

File: plugins/outputs/librato/librato.go (hunk 6 above)
Comment (zak-pawel) [reply plugins/outputs/librato/librato.go:21]: Seems it is needed for print debug message even if write has been successfully done to Librato.

File: plugins/outputs/wavefront/wavefront.go (hunk 7 above)
Comment (zak-pawel) [reply plugins/outputs/wavefront/wavefront.go]: `wavefront: ` removed from message - logger will tag the plugin name elsewhere

File: plugins/outputs/amqp/amqp.go (hunk 5 above)
Comment (zak-pawel) [reply plugins/outputs/amqp/amqp.go:6]: We can just enable and configure [imports-blacklist](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#imports-blacklist) rule for that :)

**Review by srebhan — APPROVED** [review]
Still awesome... ;-)

**Review by ssoroka — APPROVED** [review]

**Pull request merged**